cmake/checks/check_02_compiler_features.cmake

```diff
@@ -407,3 +407,19 @@ if(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
 
   reset_cmake_required()
 endif()
+
+
+#
+# Check whether the compiler supports interprocedural and link-time
+# optimization. If so, we can set the corresponding property on the
+# compile and link targets later on.
+#
+include(CheckIPOSupported)
+check_ipo_supported(RESULT _res OUTPUT _out LANGUAGES CXX)
+if (_res)
+  message(STATUS "Compiler supports interprocedural/link-time optimizations")
+  set(DEAL_II_COMPILER_SUPPORTS_IPO "YES")
+else()
+  message(STATUS "Compiler does not support interprocedural/link-time optimizations: ${_out}")
+  set(DEAL_II_COMPILER_SUPPORTS_IPO "NO")
+endif()
```

Comment:
I suggest not to enable LTO by default. It requires a lot more memory than a standard build, build times tend to increase signficantly and linking is much more fragile.

I suggest to add a  simple `DEAL_II_USE_INTERPROCEDURAL_OPTIMIZATION` option defined in `cmake/setup_cached_variables.cmake`? similar to `DEAL_II_USE_VECTORIZATION_GATHER` and dispense of the feature check altogether.

That way it can be enabled easily and we ensure that an inexperienced user doesn't have it enabled.

Reply:
What about removing the entire feature check here? It just adds unnecessary complexity - literally all compilers on Linux support link time optimization for years now, i.e., the check will always succeed.

And for the two platforms that don't support link time optimization CMake will already complain when setting the target property.